modules/ext.tabberNeue/ext.tabberNeue.js

```diff
@@ -454,207 +390,73 @@ class TabberEvent {
-		this.tablist = document.createElement( 'nav' );
-		this.indicator = document.createElement( 'div' );
+		this.tablist = tabber.querySelector( ':scope > .tabber__header > .tabber__tabs' );
 	}
 
 	/**
 	 * Sets the attributes of a tab element.
-	 *
-	 * @param {Element} tab - The tab element to set attributes for.
-	 * @param {string} tabId - The ID of the tab element.
 	 */
-	setTabAttributes( tab, tabId ) {
+	setTabsAttributes() {
 		const tabAttributes = {
-			class: 'tabber__tab',
-			href: '#' + tabId,
-			id: 'tab-' + tabId,
-			role: 'tab',
 			tabindex: '-1',
-			'aria-selected': false,
-			'aria-controls': tabId
+			'aria-selected': 'false'
 		};
-
-		Util.setAttributes( tab, tabAttributes );
-	}
-
-	/**
-	 * Creates a tab element with the given title attribute and tab ID.
-	 *
-	 * @param {string} titleAttr - The title attribute for the tab element.
-	 * @param {string} tabId - The ID of the tab element.
-	 * @return {Element} The created tab element.
-	 */
-	createTab( titleAttr, tabId ) {
-		const tab = document.createElement( 'a' );
-
-		if ( config.parseTabName ) {
-			tab.innerHTML = titleAttr;
-		} else {
-			tab.textContent = titleAttr;
+		for ( const tab of this.tablist.children ) {
+			Util.setAttributes( tab, tabAttributes );
 		}
-
-		this.setTabAttributes( tab, tabId );
-
-		return tab;
-	}
-
-	/**
-	 * Sets the attributes of a tab panel element.
-	 *
-	 * @param {Element} tabpanel - The tab panel element to set attributes for.
-	 * @param {string} tabId - The ID of the tab panel element.
-	 */
-	setTabpanelAttributes( tabpanel, tabId ) {
-		const tabpanelAttributes = {
-			id: tabId,
-			role: 'tabpanel',
-			tabindex: '-1',
-			'aria-hidden': 'true',
-			'aria-labelledby': `tab-${ tabId }`
-		};
-
-		Util.setAttributes( tabpanel, tabpanelAttributes );
 	}
 
 	/**
-	 * Creates a tab element based on the provided tab panel.
+	 * Get the active tab in init state
 	 *
-	 * @param {Element} tabpanel - The tab panel element to create a tab element for.
-	 * @return {Element|false} The created tab element, or false if the title attribute is missing
-	 * or malformed.
+	 * @param {string} urlHash - The URL hash used to set the active tab.
+	 * @return {HTMLElement}
 	 */
-	createTabElement( tabpanel ) {
-		const titleAttr = tabpanel.dataset.mwTabberTitle;
-
-		if ( !titleAttr ) {
-			mw.log.error(
-				'[TabberNeue] Missing or malformed `data-mw-tabber-title` attribute'
-			);
-			return false;
+	getActiveTab( urlHash ) {
+		const activeTab = this.tablist.firstElementChild;
+		if ( !urlHash ) {
+			return activeTab;
 		}
-
-		let tabId;
-		if ( config.parseTabName ) {
-			tabId = Util.extractTextFromHtml( titleAttr );
-		} else {
-			tabId = titleAttr;
+		const escapedHash = mw.util.escapeIdForAttribute( urlHash );
+		const idFromUrlHash = escapedHash.replace( 'tabber-tabpanel-', 'tabber-tab-' );
+		if ( idFromUrlHash === escapedHash ) {
+			return activeTab;
```

Comment:
The check is only there to check if the hash is a valid ID of a tabpanel. If idFromUrlHash === escapedHash, then it is not a valid ID of tabpanel because it does not have the prefix